src/core/geometry/qgswkbtypes.h

```diff
@@ -204,18 +208,17 @@ class CORE_EXPORT QgsWkbTypes
         case Qgis::WkbType::Unknown:
           return Qgis::WkbType::Unknown;
 
-        // until we support TIN types, use multipolygon
         case Qgis::WkbType::Triangle:
-          return Qgis::WkbType::MultiPolygon;
+          return Qgis::WkbType::TIN;
```

Comment:
This is confusing/inconsistent. If we are considering tin/polyhedral types as single types (which I agree with), then we shouldn't be returning them as the multi type here.

Reply:
Removed.